### ISSUE TITLE
segmentation fault in examples/classification.cpp

### ISSUE THREAD

**Issue by CreativeCimmons** [BVLC/caffe#3416]
Hello,

My PC config are:-
Ubuntu 14.04
opnecv 3.0.0

I was trying to run Classification.cpp but I got segmentation fault.

backtrace

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff76c49bc in cv::merge(cv::_InputArray const&, cv::_OutputArray const&) ()
   from /usr/lib/x86_64-linux-gnu/libopencv_core.so.2.4

I traced this to line 142 in classification.cpp  cv::merge(channels, mean);

Felt puzzled and replaced the call for merge with actual code (souce: https://github.com/Itseez/opencv/blob/master/modules/photo/src/merge.cpp), then I got a specific segmentation
fault images[i].convertTo(img, CV_32F, 1.0f/255.0f); (line 170 of merge.cpp)

I don't understand the error and how to rectify. I am newbie to Caffe and OpenCV . Any help would be appreciated. 

Thanks

**Comment by flx42** [comment]
Which command did you use for classification?

**Comment by CreativeCimmons** [comment]
hello flx42, thanks for responding. I didn't get you. What 'command' ?. I did a catkin build and then ran the executable...

**Comment by flx42** [comment]
Did you use the following command?
```
./build/examples/cpp_classification/classification.bin \
  models/bvlc_reference_caffenet/deploy.prototxt \
  models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel \
  data/ilsvrc12/imagenet_mean.binaryproto \
  data/ilsvrc12/synset_words.txt \
  examples/images/cat.jpg
```

**Comment by CreativeCimmons** [comment]
yes in the same intent ( Actually this were a part of rosbuild so I did a catkin_make and then rosrun package executable) I hardcoded all the paths. Same as the one you posted.

**Comment by flx42** [comment]
Are you sure your system installation is sane? You said you are using OpenCV 3.0 but apparently OpenCV 2.4 was used.

**Comment by CreativeCimmons** [comment]
"pkg-config --modversion opencv"
3.0.0

whereas dpkg -l | grep libopencv
gave out

i  libopencv-calib3d-dev:amd64                           2.4.8+dfsg1-2ubuntu1                                amd64        development files for libopencv-calib3d
ii  libopencv-calib3d2.4:amd64                            2.4.8+dfsg1-2ubuntu1                                amd64        computer vision Camera Calibration library
ii  libopencv-contrib-dev:amd64                           2.4.8+dfsg1-2ubuntu1                                amd64        development files for libopencv-contrib
.......

**Comment by CreativeCimmons** [comment]
flx42, I tried to reinstall opencv to 3.0 but after re installation, I get the same output for  dpkg -l | grep libopencv . So what you mean is that the code would work only for opencv 3.0 . If so then I have to figure out a way to remove opencv 2.4 libraries in /usr/lib and install 3.0 libraries.

**Comment by flx42** [comment]
No, the code will work for 2.4 and for 3.0. But I suspect your system is just messed up. Try removing one of the two, 3.0 seems simpler to remove in your case.
Maybe Caffe is compiled with 3.0 headers but linked against the 2.4 libraries.
